common/openapi.yml

```
@@ -20,6 +20,27 @@ paths:
           $ref: '#/components/responses/NotFound'
         '500':
           $ref: '#/components/responses/ServerError'
+  /api/v4/gcs/getSignedUrl:
+    post:
+      summary: Get a signed URL for a GCS Object, signed the caller's fence service account
```

```suggestion
      summary: Resolves the posted DRS URI and returns a signed URL for the GCS Object, signed by the caller's fence service account
```

What happens if the resolved DRS object does not have a gcs target?  For example, what if the canonical file is only stored on Azure and/or AWS?  Does a `404` returned by this endpoint mean the DRS object couldn't be found/resolved?  Or does it mean that the DRS URI was resolved but no gcs location was available to sign?

Wait, now I'm confused…looking at the request payload, this is really confusing.  I would have expected this just to be posted with a DRS URI not a project and bucket and stuff.  I'm going to read the rest of this PR to see if things make sense.  If you're reading this paragraph, it is because I forgot to come back to this comment and fix it based on what I learned reading the rest of the code

---

It does not resolve the DRS URI. It only uses it to get the right fence account key.